execution/aggregate/scalar_table.go

```diff
@@ -39,19 +40,23 @@ func newScalarTables(stepsBatch int, inputCache []uint64, outputCache []*model.S
 }
 
 func newScalarTable(inputSampleIDs []uint64, outputs []*model.Series, newAccumulator newAccumulatorFunc) *scalarTable {
-	accumulators := make([]*accumulator, len(outputs))
-	for i := 0; i < len(accumulators); i++ {
-		accumulators[i] = newAccumulator()
-	}
 	return &scalarTable{
-		inputs:       inputSampleIDs,
-		outputs:      outputs,
-		accumulators: accumulators,
+		inputs:         inputSampleIDs,
+		outputs:        outputs,
+		newAccumulator: newAccumulator,
 	}
 }
 
-func (t *scalarTable) aggregate(vector model.StepVector) {
-	t.reset()
+func (t *scalarTable) aggregate(arg float64, vector model.StepVector) {
+	if len(t.accumulators) == 0 {
+		t.accumulators = make([]*accumulator, len(t.outputs))
+		for i := 0; i < len(t.accumulators); i++ {
+			t.accumulators[i] = t.newAccumulator(arg)
+		}
+	} else {
+		t.reset()
+	}
```

Comment:
I think adding the `arg` as argument to the `accumulator reset` method and keep creating them as before - on the tables initialization)

(In theory i could even use the same accumulator instance to to another quantile calculation using a different value of `q`)

@saswatamcode WDYT?